crates/wasi-http/wit/deps/http/types.wit

```diff
@@ -202,21 +202,29 @@ interface types {
   ///
   /// These timeouts are separate from any the user may use to bound a
   /// blocking call to `wasi:io/poll.poll-list`.
-  ///
-  /// FIXME: Make this a resource to allow it to be optionally extended by
-  /// future evolution of the standard and/or other interfaces at some later
-  /// date?
-  record request-options {
+  resource request-options {
+    /// Construct a default `request-options` value.
+    constructor();
 
     /// The timeout for the initial connect to the HTTP Server.
-    connect-timeout-ms: option<u32>,
+    connect-timeout-ms: func() -> option<u32>;
+
+    /// Set the timeout for the initial connect to the HTTP Server.
+    set-connect-timeout-ms: func(ms: option<u32>);
```

Comment:
Each of these setters needs to return result, where the error means that the setting is not supported.